src/Windows/Avalonia.Win32/DxgiSwapchain/DxgiRenderTarget.cs

```diff
@@ -0,0 +1,188 @@
+        }
+
+        public override IGlPlatformSurfaceRenderingSession BeginDraw()
+        {
+            if (_swapChain is null)
+            {
+                throw new InvalidOperationException("No chain to draw on");
+            }
+
+            var contextLock = _egl.PrimaryContext.EnsureCurrent();
+            EglSurface? surface = null;
+            IDisposable? transaction = null;
+            var success = false;
+            HRESULT retval = default;
+            try
+            {
+                Interop.UnmanagedMethods.RECT pClientRect;
+                GetClientRect(_window.Handle, out pClientRect);
+                if (!RectsEqual(pClientRect, _clientRect))
+                {
+                    // we gotta resize 
+                    _clientRect = pClientRect;
+
+                    if (_renderTexture is not null)
+                    {
+                        _renderTexture.Dispose();
+                        _renderTexture = null;
+                    }
+
+                    _swapChain.ResizeBuffers(2,
+                        (ushort)(pClientRect.right - pClientRect.left),
+                        (ushort)(pClientRect.bottom - pClientRect.top),
+                        DXGI_FORMAT.DXGI_FORMAT_B8G8R8A8_UNORM,
+                        (ushort)_flagsUsed
+                        );
+                }
+
+                var size = _window.Size;
+
+                // Get swapchain texture here 
+                var texture = _renderTexture;
+                if (texture is null)
+                {
+                    Guid textureGuid = ID3D11Texture2DGuid;
+                    texture = MicroComRuntime.CreateProxyFor<IUnknown>(_swapChain.GetBuffer(0, &textureGuid), true);
+                }
+                _renderTexture = texture;
+
+                // I also have to get the pointer to this texture directly 
+                surface = ((AngleWin32EglDisplay)_egl.Display).WrapDirect3D11Texture(_egl, MicroComRuntime.GetNativeIntPtr<IUnknown>(_renderTexture),
+                    0, 0, size.Width, size.Height);
+
+                var res = base.BeginDraw(surface, _window, () =>
+                {
+                    _swapChain.Present((ushort)0U, (ushort)0U);
+                    surface?.Dispose();
+                    transaction?.Dispose();
+                    contextLock?.Dispose();
+                }, true);
+                success = true;
+                return res;
+            }
+            finally
+            {
+                if (!success)
+                {
+                    surface?.Dispose();
+                    if (_renderTexture is not null)
+                    {
+                        _renderTexture.Dispose();
+                        _renderTexture = null;
+                    }
+                    transaction?.Dispose();
+                    contextLock.Dispose();
+                }
+            }
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
```

Comment:
Fields should be disposed too.

Reply:
DxgiRenderTarget now disposes of its fields when it is disposed.